packages/cli/src/modules/insights/insights.service.ts

```diff
@@ -33,25 +34,45 @@ export class InsightsService {
 		private readonly collectionService: InsightsCollectionService,
 		private readonly pruningService: InsightsPruningService,
 		private readonly licenseState: LicenseState,
+		private readonly instanceSettings: InstanceSettings,
 		private readonly logger: Logger,
 	) {
 		this.logger = this.logger.scoped('insights');
 	}
 
 	startTimers() {
-		this.compactionService.startCompactionTimer();
+		if (
+			this.instanceSettings.instanceType !== 'main' &&
+			this.instanceSettings.instanceType !== 'webhook'
+		) {
+			return;
+		}
+
 		this.collectionService.startFlushingTimer();
+		this.logger.debug('Started collection flushing scheduler');
+
+		// Start compaction and pruning timers for main leader instance only
+		if (this.instanceSettings.isLeader) {
+			this.startCompactionAndPruningSchedulers();
+		}
+	}
+
+	@OnLeaderTakeover()
+	startCompactionAndPruningSchedulers() {
```

Comment:
Sometimes we call them schedulers and sometimes timers. No preference, but let's stay consistent?